### ISSUE TITLE
enhance 'ipfs diag sys' output

### ISSUE THREAD

**Issue by RubenKelevra** [ipfs/kubo#7458]
We like to extend the `ipfs diag sys` output to include more system information than just the platform, architecture, and the go version.

Currently, a typical output looks like this:

```
{
  "diskinfo": {
    "free_space": 18522890240,
    "fstype": "2435016766",
    "total_space": 14918279168
  },
  "environment": {
    "GOPATH": "",
    "IPFS_PATH": ""
  },
  "ipfs_commit": "",
  "ipfs_version": "0.6.0-dev",
  "memory": {
    "swap": 15964000,
    "virt": 5207176000
  },
  "net": {
    "interface_addresses": [
      "/ip4/127.0.0.1",
      "/ip4/192.168.8.2",
      "/ip6/::1",
      "/ip6/fe80::8c71:2b0e:6947:3090"
    ],
    "online": true
  },
  "runtime": {
    "arch": "amd64",
    "compiler": "gc",
    "gomaxprocs": 4,
    "numcpu": 4,
    "numgoroutines": 1864,
    "os": "linux",
    "version": "go1.14.4"
  }
}
```

@Stebalien likes to remove the interface_addresses from the output.

I like to add some more information, like those commands provide:

```
Linux:
uname -a; grep -Hv "^#" /etc/*release
BSD:
uname -a; uname -K
MacOS:
uname -a; sw_vers
```

_This idea was discussed earlier in the PR https://github.com/ipfs/go-ipfs/pull/7408_

**Comment by Stebalien** [comment]
Note: The implementation of this command shouldn't invoke external commands if at all possible. Most of this information can be read from files in the filesystem (or even from go APIs).

**Comment by RubenKelevra** [comment]
Yeah. We might want to collect also if the machine is a virtual machine (and which kind of virtualization) or bare metal, since this might have an impact on the performance.

Thinking that we probably get more and more user feedback in the future, when Filecoin gets a lot of attention, so I try to look into the documentations to spot errors or uncertainties for newcomers at the moment.

I think getting all informations at once with one command would save a round-trip per ticket and some time to write the first answer, and reread the issue one day later with the additional informations.

So it makes sense to brainstorm some kind of wishlist or common questions.

I think common questions are:

* Are there experimental options activated?
* Which status has autonat?
* Which datastore is used?
* How many data/objects are stored in the datastore?
* Which Linux distribution?
* Which Kernel?
* Is QUIC active/deactivated?
* Which is the order of the transports (Noise, TLS, Secio) or which are active?

Additionally I think we need a way to easily figure out which kind of transport-protocol is currently used between two peers, this could help to pinpoint performance issues on one specific protocol.
